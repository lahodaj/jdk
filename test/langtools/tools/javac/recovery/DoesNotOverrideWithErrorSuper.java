/**
 * @test /nodynamiccopyright/
 * @bug 8282160
 * @summary Prevent method does not override a super type method.
 * @compile/fail/ref=DoesNotOverrideWithErrorSuper.out -XDrawDiagnostics DoesNotOverrideWithErrorSuper.java
 */
public class DoesNotOverrideWithErrorSuper {
    public static class UnknownExtends extends Unknown {
        @Override public String toString() { return null; }
        @Override public void test() { }
    }
    public static class UnknownImplements implements Unknown {
        @Override public String toString() { return null; }
        @Override public void test() { }
    }
    public static interface UnknownImplementsIntf extends Unknown {
        @Override public default void test() { }
    }
    public static class UnknownExtendsBase extends Unknown {}
    public static class UnknownExtends2 extends UnknownExtendsBase {
        @Override public String toString() { return null; }
        @Override public void test() { }
    }
    public static interface UnknownImplementsBase extends Unknown {}
    public static class UnknownImplements2 implements UnknownImplementsBase {
        @Override public String toString() { return null; }
        @Override public void test() { }
    }
}
