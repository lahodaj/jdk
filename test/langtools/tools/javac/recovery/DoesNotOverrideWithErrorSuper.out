DoesNotOverrideWithErrorSuper.java:8:48: compiler.err.cant.resolve.location: kindname.class, Unknown, , , (compiler.misc.location: kindname.class, DoesNotOverrideWithErrorSuper, null)
DoesNotOverrideWithErrorSuper.java:12:54: compiler.err.cant.resolve.location: kindname.class, Unknown, , , (compiler.misc.location: kindname.class, DoesNotOverrideWithErrorSuper, null)
DoesNotOverrideWithErrorSuper.java:16:59: compiler.err.cant.resolve.location: kindname.class, Unknown, , , (compiler.misc.location: kindname.class, DoesNotOverrideWithErrorSuper, null)
DoesNotOverrideWithErrorSuper.java:19:52: compiler.err.cant.resolve.location: kindname.class, Unknown, , , (compiler.misc.location: kindname.class, DoesNotOverrideWithErrorSuper, null)
DoesNotOverrideWithErrorSuper.java:24:59: compiler.err.cant.resolve.location: kindname.class, Unknown, , , (compiler.misc.location: kindname.class, DoesNotOverrideWithErrorSuper, null)
5 errors
